srml/metadata/src/lib.rs

```diff
@@ -42,6 +42,9 @@ use rstd::vec::Vec;
 #[cfg(feature = "std")]
 type StringBuf = String;
 
+/// Curent version of metadata
+pub const META_VERSION: u32 = 0x6174656d; // 'meta' warn endianness
```

Comment:
I'm not 100%, however you probably know better - if we encode as str, we probably add the length as well. It is actually quite easy (as done here), when it is a u32 to just read and compare. (Basically in the raw stream `meta` will show.)

So the decoders can read the start as u32 (as defined here), and it does not match, just bail. The same can be done with string, however if it is invalid and the length prefix is something weird, it may just be slightly worse since it becomes basically a variable-length-decoding.

Either way, up to you guys - will conform to whatever is decided.

Reply:
No you both are right. I did not thought about the added length for the string.